src/coreclr/vm/codeman.h

```diff
@@ -2449,6 +2627,141 @@ class ReadyToRunJitManager final: public IJitManager
 
 #endif
 
+#ifdef FEATURE_INTERPRETER
+
+class InterpreterJitManager final : public EECodeGenManager
+{
+    VPTR_VTABLE_CLASS(InterpreterJitManager, EECodeGenManager)
+public:
+    ICorJitCompiler *   m_interpreter;
+    HINSTANCE           m_interpreterHandle;
+
+    InterpreterJitManager();
+
+    virtual ICorJitCompiler* GetCompiler()
+    {
+        return m_interpreter;
+    }
+
+    virtual ICorJitCompiler* GetAltCompiler()
+    {
+        return NULL;
+    }
+
+    BOOL LoadInterpreter();
+
+    BOOL IsInterpreterLoaded()
+    {
+        LIMITED_METHOD_CONTRACT;
+        return m_interpreter != NULL;
+    }
+
+    virtual DWORD GetCodeType()
+    {
+        LIMITED_METHOD_DAC_CONTRACT;
+        return (miManaged | miIL);
```

Comment:
Is this what we want? Grepping around it seems like some callers will assume that `(miManaged | miIL)` means JITcode, and the DAC will also treat this as JITcode I think. Vs if we returned something else the DAC would see it as UNKNOWN (what we want I think?)

i.e. `PublishUnwindInfoForExistingMethods` looks for this, I'm not sure whether interpreter IR would strictly need unwind info in the same way or if it would just be polluting a table that should only contain native executable code (JIT or R2R, regardless) that will be on the stack during an unwind?

Reply:
Interpreter won't use any unwind info and the PublishUnwindInfoForExistingMethods is not called for the interpreter code. 
It seems that the miManaged and miIL needs to be there as it is both correct. I was thinking about adding some extra flag for interpreter code, but it is problematic as the code type has only two bits reserved and the CorMethodImpl is something that ILASM sets too. So I assume C# compiler would also set it. I didn't want to require tools to need to  be modified.
So far, I haven't hit any issue with the current state, but it is theoretically possible that we will need to revisit this eventually. I'll add a TODO here.

Reply:
> it is problematic as the code type has only two bits reserved and the CorMethodImpl is something that ILASM sets too.

This method is only used in DAC and in few asserts. It is an anti-parent that it reuses CorMethodImpl enum that is part of the ECMA metadata definition. The value returned by this method has nothing to do with the ECMA metadata. We can change it to use its own enum with as many bits as we need if necessary.